server/controllers/allocationsController.ts

```diff
@@ -128,76 +131,75 @@ export default class AllocationsController {
     })
   }
 
+  // eslint-disable-next-line consistent-return
   async choosePractitioner(req: Request, res: Response, crn, convictionNumber, teamCode) {
-    const { token } = res.locals.user
-    const { offenderManagers } = await this.workloadService.getOffenderManagersToAllocate(token, teamCode)
-    const { name: teamName } = await this.probationEstateService.getTeamDetailsByCode(token, teamCode)
+    const { token, username } = res.locals.user
 
-    const offenderManagersToAllocate = offenderManagers
-      .map(
-        om =>
-          new AllocateOffenderManager(
-            om.forename,
-            om.surname,
-            om.grade,
-            om.capacity,
-            om.totalCommunityCases,
-            om.totalCustodyCases,
-            om.code,
-            om.totalCasesInLastWeek,
-            om.email
-          )
-      )
-      .sort((a: AllocateOffenderManager, b: AllocateOffenderManager) => {
-        if (b.gradeOrder === a.gradeOrder) {
-          return a.capacity - b.capacity
-        }
-        return b.gradeOrder - a.gradeOrder
-      })
+    const teamCodesPreferences = await this.userPreferenceService.getTeamsUserPreference(token, username)
 
-    const missingEmail = offenderManagersToAllocate.some(i => !i.email)
-    const response: CaseForChoosePractitioner = await this.allocationsService.getCaseForChoosePractitioner(
-      token,
-      crn,
-      convictionNumber
+    const [allocationInformationByTeam, allTeamDetails] = await Promise.all([
```

Comment:
This is the "meat" of the change. We now get all data from workload then loop thru the teams to generate the tab data